hw/ip/entropy_src/doc/_index.md

```diff
@@ -46,28 +50,41 @@ However, as long as the function provided by the register set supports all of th
 
 # Theory of Operations
 
-As already described, this IP block will collect bits of entropy for firmware consumption.
-This revision is a straightforward implementation using an LFSR.
-In principal, any polynomial may be used so long as it is primitive in Galois Field of order 2, or GF(2).
+As already described, this IP block will collect bits of entropy for firmware or hardware consumption.
+This revision supports both an LFSR for the digital implementation, and an external interface for the analog implementation.
+For the LRSR implementation, any polynomial may be used so long as it is primitive in Galois Field of order 2, or GF(2).
 However, preference should be given to polynomials with the fewest number of terms.
 
-The first step is enabling and initialization.
-The main difference between these steps is that enabling is a global block enable.
-The initialization step can be done at anytime while the block is enabled and running.
-Before setting the `ENABLE` bit, the LFSR seed input should be initialized, using the {{< regref "ES_SEED" >}} register.
+The first step is initialization and enabling.
+Before setting the `ENABLE` field, the LFSR seed input should be initialized, using the {{< regref "ES_SEED" >}} register.
 The {{< regref "ES_SEED" >}} register will transfer its value according to the initialization sequence.
 After the block is enabled and initialized, entropy bits will be collected up indefinitely until disabled.
+If analog mode is desired, the `ANA_SRC_EN` bit should be set first.
+Once the `es_ana_src_ok` interrrupt or status bit is active, then the `ENABLE` field can be set to enable analog mode.
 
 It is assumed that any entropy source will drop bits.
-This means that the FIFO used to collect up the entropy bits will fill, and until the firmware starts pulling out bits, continuously generated entropy source bits may be dropped from usage.
-The process of filling and draining the FIFO is the same, independent of what speed the entropy generation rate is at.
+This means that the FIFO used to collect up the entropy bits will fill, and until the firmware or hardware starts pulling out bits, continuously generated entropy source bits may be dropped from usage.
```

Comment:
@cdgori 
just to clarify this point here..do you mean if the downstream doesn't drain, once the FIFOs in entropy src fill up, the module more or less becomes static? If yes, I'm trying to think through if this is a good idea.. since you have two quarantine FIFOs, I honestly wonder if we should just constantly drain and fill.  Ie, when one quarantine fifo is marked available, it should simply deposit its contents into the TLUL and HW fifos.  If those happen to be full, just pop-off a "stale" entry and accept the new entropy.  This way things are always changing.

I hope Chris can chime in with his thoughts there.

Reply:
That operation is certainly doable for the FIFOs. I was thinking that we need to handle the "bursty" case with the quarantine fifos backing up the more shallow HW and TLUL fifos. If the quarantine fifos are to continuously fill and drop, then the other fifos may have to be deeper. I would like to get a "yes/no" on this behavior change.

Reply:
If "stale" entries are popped off and new entropy fills the FIFOs, For dv, I would need to be able to predict when this happens via a signal or time period or something. Mark talked about maybe having "static/dynamic" be controlled via a mode bit but that adds another mode to verify. Do we know which way it should be so we don't need to add a mode?

Reply:
Sorry I did not comment directly here, but we ultimately talked about it on the call a lot.  

I think this section of the design will be getting slightly reworked now, based on the updated spec (?) but I believe that on the call we ended up saying it was OK to spill one of the quarantine FIFOs if everything was full but that it wouldn't strictly be necessary.  (i.e. freshness didn't degrade by holding the entropy in the FIFO,)

(I believe the main concern was the instantaneous/burst fill capability right after an invalidate, and therefore if the quarantine would have to be double-sized to deal with that worst-case condition.)